2021-05-06-updating-via-direct-commands-no-longer-recommended.md

```diff
@@ -0,0 +1,36 @@
+---
+layout: post
+title: "Updating via direct commands no longer recommended"
+categories:
+ - security
+ - announcements
+author: Andrew David Wong
+---
+
+For many years now, Qubes users have been accustomed to updating their systems using commands like `qubes-dom0-update`, `dnf update`, and `apt update` (along with several variants). Recently, however, the Qubes team has made numerous security improvements to the way updates work in Qubes OS (see below for examples). In order to benefit from these improvements, it is important that users switch from using the old direct commands to using the **Qubes Update** tool or its command-line equivalents, which are documented in our newly-revised [Updating Qubes OS](/doc/updating-qubes-os/) page.
+
+_**Note:** This change is only about **updating**. It's still fine to use direct commands for **installing** new packages._
+
+## User action summary
+
+ - **Do** keep updating your system frequently, including dom0, TemplateVMs, and StandaloneVMs (if you have any).
+ - **Avoid** direct use of old commands like `qubes-dom0-update`, `dnf update`, and `apt update` (and their variants).
+ - **Do** use the **Qubes Update** tool (or its `qubesctl` command-line equivalents), as documented in [Updating Qubes OS](/doc/updating-qubes-os/).
+
+## Security explanation and examples
+
+When you update using the **Qubes Update** tool (or its `qubesctl` command-line equivalents), you are updating via the [Salt management stack](/doc/salt/). Salt allows the Qubes developers to deliver security fixes that cannot be handled by normal package updates. For example, you can think of updating a Fedora TemplateVM with the **Qubes Update** tool as doing `dnf update` *plus* any available Salt configuration steps from the Qubes developers. In some cases, certain security bugs can only be fixed via these Salt configuration steps. They can't be handled by a package installed via `dnf` or another package manager alone.
+
+There are four main examples that illustrate the importance of updating via Salt:
+
+1. [QSB-046 / DSA-4371-1: APT update mechanism vulnerability](/news/2019/01/23/qsb-46/). Since the security flaw was in the APT update mechanism itself, the alternative patching method described in QSB-046 relied on using special Salt configuration steps to fix it. It was very important to fix APT updates *before* attempting to use APT to update normally.
```

Comment:
The Qubes manager also included the fix, BTW.  It should do so for future fixes (though see erratum below).

Reply:
Good to know, thanks, but I don't think it's important to get into that level of historical detail in the context of this post. (Let me know if you disagree.)